libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -341,7 +341,7 @@ string ABIFunctions::conversionFunction(Type const& _from, Type const& _to)
 				solAssert(_from.mobileType(), "");
 				body =
 					Whiskers("converted := <cleanEnum>(<cleanInt>(value))")
-					("cleanEnum", cleanupFunction(_to, false))
```

Comment:
Why not pass along the flag?

Reply:
There's no flag in the "conversionFunction" to be passed around.

Reply:
I hope we can use that function for general conversions. The decoder always verifies prior to conversion, right?

Reply:
I hope this conversion function can later be used for the solidity to yul code generator.

My question would be: If we assert in a failed conversion here. Might it be that the ABI decoder calls convert to enum (from whatever) without validating the range prior to calling convert? If not, we should revert instead.